### ISSUE TITLE
[Bug]: Can't login with Chrome 61

### ISSUE THREAD

**Issue by presslab-us** [nextcloud/server#37311]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I have an Android 8 device which is using Chrome Webview 61.0.3163.79.  Using the Nextcloud app, I cannot enter the credentials to login, I just see the background image but no credential entry form.

I've installed the same version Chrome on my PC so I can debug the issue.  These are the errors I am seeing:

```
core-common.js?v=35939c90-0:2 Uncaught SyntaxError: Unexpected token {
fileinfo.js:181 Uncaught ReferenceError: OC is not defined
    at fileinfo.js:181
(anonymous) @ fileinfo.js:181
merged-template-prepend.js?v=35939c90-0:26 Uncaught ReferenceError: OC is not defined
    at merged-template-prepend.js?v=35939c90-0:26
(anonymous) @ merged-template-prepend.js?v=35939c90-0:26
index.js:25 Uncaught ReferenceError: OC is not defined
    at index.js:25
(anonymous) @ index.js:25
apporder.js?v=35939c90-0:1 Uncaught ReferenceError: $ is not defined
    at apporder.js?v=35939c90-0:1
(anonymous) @ apporder.js?v=35939c90-0:1
script.js?v=35939c90-0:492 Uncaught ReferenceError: jQuery is not defined
    at script.js?v=35939c90-0:492
(anonymous) @ script.js?v=35939c90-0:492
files.js?v=35939c90-0:122 Uncaught ReferenceError: jQuery is not defined
    at files.js?v=35939c90-0:122
(anonymous) @ files.js?v=35939c90-0:122
theming.js?v=35939c90-0:1 Uncaught ReferenceError: OCA is not defined
    at theming.js?v=35939c90-0:1
(anonymous) @ theming.js?v=35939c90-0:1
```

I've tracked down the first error in `dist/core-common.js` to a catch statement.  Looking at the code I see this: `...catch{return!1}...`

I then executed `sed -i 's/catch{/catch(e){/g' core-common.js` which adds the (e) parameter to all of the catch statements.  **This is working now, I can see the login form and the Nextcloud app is working.**

I don't know what files are used to generate `core-common.js`. It seems like the older version of Chrome requires a parameter on the catch statement.

### Steps to reproduce

1. Install Chrome 61: (https://www.slimjet.com/chrome/download-chrome.php?file=win%2Fchrome64_61.0.3163.79.exe)
2. Open Nextcloud login page
3. No credential entry form is shown

### Expected behavior

Credential form is shown and able to login

### Installation method

None

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "cloud.domain.tld"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "25.0.4.1",
        "overwrite.cli.url": "https:\/\/cloud.domain.tld\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "upgrade.disable-web": true,
        "log_type": "file",
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "loglevel": 2,
        "log.condition": {
            "apps": [
                "admin_audit"
            ]
        },
        "mail_smtpmode": "smtp",
        "remember_login_cookie_lifetime": "1800",
        "log_rotate_size": "10485760",
        "trashbin_retention_obligation": "auto, 60",
        "versions_retention_obligation": "auto, 180",
        "activity_expire_days": "120",
        "simpleSignUpLink.shown": false,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0.5,
            "dbindex": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "session_lifetime": 604800,
        "default_phone_region": "us",
        "logtimezone": "America\/Vancouver",
        "htaccess.RewriteBase": "\/",
        "share_folder": "\/Shared",
        "enable_previews": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\Movie",
            "OC\\Preview\\Photoshop",
            "OC\\Preview\\SVG",
            "OC\\Preview\\TIFF",
            "OC\\Preview\\HEIC"
        ],
        "preview_max_x": "3840",
        "preview_max_y": "2160",
        "preview_max_memory": "4096",
        "preview_max_filesize_image": "256",
        "jpeg_quality": "60",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "app_install_overwrite": [
            "extract",
            "pdfannotate",
            "integration_whiteboard",
            "ojsxc",
            "breezedark",
            "apporder"
        ],
        "csrf.optout": [
            "\/^WebDAVFS\/",
            "\/^Microsoft-WebDAV-MiniRedir\/",
            "\/KeeWeb\\\/\\d+\\.\\d+\\.\\d+\/"
        ],
        "updater.release.channel": "beta",
        "has_rebuilt_cache": true,
        "memories.exiftool": "\/var\/www\/nextcloud\/apps\/memories\/exiftool-bin\/exiftool-amd64-glibc",
        "memories.gis_type": 2
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - admin_audit: 1.15.0
  - apporder: 0.15.0
  - calendar: 4.3.0-rc.2
  - camerarawpreviews: 0.8.1
  - carnet: 0.24.6
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contacts: 5.2.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - externalportal: 1.0.2
  - extract: 1.3.5
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_external: 1.17.0
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - gpxpod: 5.0.6
  - keeweb: 0.6.10
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - mail: 2.3.0-alpha.4
  - maps: 1.0.0
  - memories: 4.12.3
  - metadata: 0.17.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - notify_push: 0.6.0
  - oauth2: 1.13.0
  - password_policy: 1.15.0
  - previewgenerator: 5.2.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recognize: 3.7.0
  - recommendations: 1.4.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - support: 1.8.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - bruteforcesettings
  - deck: 1.8.3
  - encryption
  - files_antivirus: 4.0.2
  - files_inotify
  - issuetemplate: 0.7.0
  - ojsxc: 5.0.0
  - photos: 2.0.1
  - survey_client: 1.13.0
  - suspicious_login
  - twofactor_totp
  - unsplash: 2.2.0
  - user_ldap
```

**Comment by joshtrichards** [comment]
This is Optional catch binding - which wasn't supported in Chrome until v66:

https://chromestatus.com/feature/5810795665424384

The requirements doc and/or the frontend code should probably be updated to require >=v66.

I poked around and there are some other spots where there could be problems like this - though it's important to note that just fixing the issue in the *login* workflow will at least permit older devices to login (and the NC Android app will pick up much of the slack after that - though probably not all of it).

The question is which path is appropriate:

1. The code be left as-is in the frontend and the requirements doc be updated to not support older browser versions?
https://docs.nextcloud.com/server/latest/admin_manual/installation/system_requirements.html#desktop-client
2. Or should the code be adjusted in the frontend and a recommendation to use a newer browser version added to the docs (for the time being)?
3. Or should both 1 and 2 be implemented (for the time being)?

Oh, and also how much does this come up? v66 is pretty old! So that might inform things.

I personally propose a compromise: a doc update with STRONG recommendation to use a semi-current browser release + login page code adjustments if straightforward enough in order to permit broad compatibility at least on the login page. Also, a warning if an older browser is detected on the login page might be a cleaner user experience.

I poked around a bit and this isn't coming from where I'd expect it to be in the code. I suspect it's in some dependencies (which I haven't dug into yet). I'm not familiar with the dependencies on the login process in the front-end.

Btw @presslab-us this is a phenomenal bug report. :)

**Comment by joshtrichards** [comment]
Someone else encountered this issue in the past week in the android app client repo: https://github.com/nextcloud/android/issues/11427

**Comment by presslab-us** [comment]
@joshtrichards Thanks for looking into this!

Of course I would like option 2.  :)

I tried to upgrade the Webview on this Android device but I was not able to change it.  It's actually a currently selling product here: https://www.hammerhead.io/products/hammerhead-karoo-2.  I use the Nextcloud app for synchronizing the track logs, and it's working great now (with my small fix).

It is certainly an old OS and I understand the desire to not keep baggage in the code for old browsers.  But as you mention keeping the login page operational is a goal that's probably good enough, and in this case adding the (now optional) catch parameter is not a big change to the code.  However I understand that adding this change to dependencies could be difficult.

**Comment by juliushaertl** [comment]
Let's close this. Chrome 61 is quite old and Nextcloud 25 is EOL since October 2023. If there is still an issue with the latest releases, please open a new issue.

**Comment by presslab-us** [comment]
tl;dr For anyone with this same problem, I have found that the FolderSync Android app (tested v4.1.1) can sync to the latest Nextcloud server using my Karoo device with it's ancient Android version.  I paid for the single user license and downloaded the APK directly from their web site.

Unfortunately I crashed on my bike and cracked the screen so I bought a new Karoo device.  But this new one still has the ancient version of Android and could not see the login screen.  I tried my old tricks above and there were now many other changes in Nextcloud that were not so easy to work around.

Anyway I tried a bunch of sync apps and they had a variety of problems.  Thankfully I did find one that works!
